lib/client/conntest/database/postgres.go

```diff
@@ -0,0 +1,108 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package database
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/gravitational/trace"
+	"github.com/jackc/pgconn"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	// A simple query to execute when running the Ping request
+	selectOneQuery = "select 1;"
+)
+
+// PostgresPinger implements the DatabasePinger interface for the Postgres protocol
+type PostgresPinger struct{}
+
+// Ping connects to the database and issues a basic select statement to validate the connection.
+func (p PostgresPinger) Ping(ctx context.Context, req PingRequest) error {
+	if err := req.CheckAndSetDefaults(); err != nil {
+		return trace.Wrap(err)
+	}
+
+	pgconnConfig, err := pgconn.ParseConfig(
+		fmt.Sprintf("postgres://%s@%s:%d/%s",
+			req.Username,
+			req.Host,
+			req.Port,
+			req.Database,
+		),
+	)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	conn, err := pgconn.ConnectConfig(ctx, pgconnConfig)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	defer func() {
+		if err := conn.Close(ctx); err != nil {
+			logrus.WithError(err).Info("failed to close connection in PostgresPinger.Ping")
+		}
+	}()
+
+	result, err := conn.Exec(ctx, selectOneQuery).ReadAll()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	if len(result) != 1 {
+		if err != nil {
+			return trace.Errorf("unexpected length for result: %+v", result)
+		}
```

Comment:
The `if err != nil {` check seems to be not needed:
```suggestion
		return trace.Errorf("unexpected length for result: %+v", result)

```